OAuth2/AuthorizationRoot.cs

```diff
@@ -69,7 +69,8 @@ public virtual IEnumerable<IClient> Clients
         /// </summary>        
         protected virtual IEnumerable<Type> GetClientTypes()
         {
-            return Assembly.GetExecutingAssembly().GetTypes().Where(typeof (IClient).IsAssignableFrom);
+          var clientType = typeof(IClient);
+          return AppDomain.CurrentDomain.GetAssemblies().SelectMany(s => s.GetTypes()).Where(p => clientType.IsAssignableFrom(p));
```

Comment:
You would just need to make a commit with your changes to your fork and it should be picked up here.  I think your commit is fine.. if It becomes a performance problem we can change it later.